compiler/rustc_middle/src/ty/util.rs

```diff
@@ -1751,6 +1751,11 @@ pub fn reveal_opaque_types_in_bounds<'tcx>(
     tcx: TyCtxt<'tcx>,
     val: ty::Clauses<'tcx>,
 ) -> ty::Clauses<'tcx> {
+    // No need to reveal opaques with the new solver enabled.
+    if tcx.next_trait_solver_globally() {
+        return val;
+    }
```

Comment:
vibeck: why here and not in `ParamEnv::with_reveal_all_normalized`?

Reply:
Ya i can do it there and then just assert here

Reply:
r=me after that